src/Avalonia.Animation/Animator`1.cs

```diff
@@ -37,12 +37,29 @@ public virtual IDisposable Apply(Animation animation, Animatable control, IObser
              if (!_isVerifiedAndConverted)
                 VerifyConvertKeyFrames();
 
-            return match
+            var matchStream = match
```

Comment:
I _think_ there might be an easier way to do this:

```
            return match
                .DistinctUntilChanged()
                .Select(x => x ? RunKeyFrames(animation, control, onComplete) : null)
                .Buffer(2, 1)
                .Where(x => x.Count > 1)
                .Subscribe(x => x[0]?.Dispose());
```

What that should do is create an old/new pair of `IDisposable`s and if the old value is non-null, call `Dispose` on it. Does that seem right to you?

Reply:
Never knew about `Buffer()` before, i learned something today :smile: I'll test that one out sir